### PR TITLE
chore: enable importas for lint

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: tools/linter/golangci-lint/.golangci.yml [hunk 1]
```diff
@@ -37,6 +38,32 @@ linters-settings:
...
+      - pkg: github.com/envoyproxy/gateway/api/v1alpha1
+        alias: egv1a1
+      # kubernetes api
+      - pkg: k8s.io/apimachinery/pkg/apis/meta/v1
+        alias: metav1
+      - pkg: k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1
+        alias: apiextensionsv1
+      - pkg: sigs.k8s.io/mcs-api/pkg/apis/v1alpha1
+        alias: mcsapiv1a1
+      - pkg: k8s.io/api/(\w+)/(v[\w\d]+)
```
Comment (arkodg) [inline tools/linter/golangci-lint/.golangci.yml:65]: what's this extra one

Comment (shawnh2) [reply tools/linter/golangci-lint/.golangci.yml:65]: match in regexp, like matching `k8s.io/api/policy/v1`, the alias will be `policyv1`

Comment (arkodg) [reply tools/linter/golangci-lint/.golangci.yml:65]: ah cool

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by cnvergence — APPROVED** [review]

**Pull request merged**